core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -972,16 +972,16 @@ class SnappySession(@transient private val sc: SparkContext,
             }
           case _ => // ignore
         }
-        cacheManager.uncacheQuery(Dataset.ofRows(this, plan))
-        if (sessionCatalog.isTemporaryTable(tableIdent)) {
-          // This is due to temp table
-          // can be made from a backing relation like Parquet or Hadoop
-          sessionCatalog.unregisterTable(tableIdent)
-        } else {
-          sessionCatalog.unregisterDataSourceTable(tableIdent, Some(br))
-        }
         br match {
           case d: DestroyRelation => d.destroy(ifExists)
+            cacheManager.uncacheQuery(Dataset.ofRows(this, plan))
+            if (sessionCatalog.isTemporaryTable(tableIdent)) {
+              // This is due to temp table
+              // can be made from a backing relation like Parquet or Hadoop
+              sessionCatalog.unregisterTable(tableIdent)
+            } else {
+              sessionCatalog.unregisterDataSourceTable(tableIdent, Some(br))
+            }
```

Comment:
Can you tell why has this code block moved here? The idea of original was to able to drop normal EXTERNAL and TEMPORARY tables with the drop table command too. With this change it will no longer be possible.